source/authentication/email-password.txt

```diff
@@ -416,15 +410,15 @@ field in the object returned by the custom password reset function:
    * - ``success``
 
      - {+service-short+} changes the user's password to the provided ``password``
-       parameter immediately. Only recommended if your application can
-       authenticate the user in the custom function itself through some
-       secure means, such as a security question.
+       parameter immediately. This can be dangerous. Only recommended if there is 
+       some level of authentication within the custom function. This authentication 
+       should be secure; for example, a security question.
 
    * - ``pending``
 
-     - {+service-short+} waits for client application to reset the user's password
-       using the Client SDK ``resetPassword`` function. The user's password is
-       not affected unless the client application calls the ``resetPassword``
+     - {+service-short+} waits for client application to reset the user's password.
+       This is more secure, as it uses the Client SDK ``resetPassword`` function. 
```

Comment:
ohh okay this makes more sense! I think I was confused as to what exactly this function was doing. is there a place i can/should get more information on what this pending status means, or just rewrite the bullet based on what you've provided here?

Reply:
Honestly, I'm not sure what the right thing to say here is. The existing bullet before your edits also doesn't make a lot of sense to me so I think it could use some additional clarification. I would probably ask in the team channel and see if anyone there can explain it. Unfortunately the git blame on the code example below this is not terribly helpful since we moved the repo; we could go back and check it out in docs-realm to find out who wrote the code example but it would be a bit of work. My guess would be Nick, so he might be a good starting point, but I think many folks on the team might be able to make an educated guess based on the code example.

I suspect we should clarify above the table that because the developer reading the documentation is the one who writes this custom function, the logic behind these values can be anything you want them to be. But the _effect_ of the status is that App Services does or doesn't reset the password. When the func returns `success`, App Services resets the password. When the func returns `pending` or `fail`, App Services does not reset the password. Pending is probably just there to let the developer know it hasn't failed, but is waiting for some additional validation.

Reply:
hi!! so i talked to ben and we reworked this together. for the specific issue you brought up, it seems like all the sdks *should* have a resetPassword function (for ex, the [node.js sdk page](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCS23228/sdk/node/examples/manage-email-password-users/)), and either the swift documentation is lacking, or this functionality doesn't work for swift, in which case we need to add a note to this section to say it doesn't work for swift

Reply:
Aha! That's the issue - I just checked all of the other SDKs and none of them besides Node/RN have documented the `resetPassword` func, but all of them have it in the API. So this is a separate docs improvement we need to make. I'll put in Jira tickets for that.

Thanks for clarifying with Ben and following up!

Reply:
Hah, I started to add Jira tickets to address this as missing docs for Swift/Java/.Net, but I actually think there is a misunderstanding here about the `resetPassword` func. I'll bring it up after standup tomorrow to try to get to the bottom of it.

Reply:
sounds good!! as for this PR, am I good to have you review it/put it up for general LFR, or should we wait for after standup?

Reply:
I'll take a look in the morning at your revisions!